### DISCUSSION TITLE
Isaac Lab 1.4.0 is out!

### DISCUSSION THREAD

**Discussion by rthaker01** [isaac-sim/IsaacLab#1572] [Announcements]
👀 Overview
Due to a great amount of amazing updates, we are putting out one more Isaac Lab release based off of Isaac Sim 4.2. This release contains many great new additions and bug fixes, including several new environments, distributed training and hyperparameter support with Ray, new live plot feature for Manager-based environments, and more.

We will now spend more focus on the next Isaac Lab release geared towards the new Isaac Sim 4.5 release coming soon. The upcoming release will contain breaking changes in both Isaac Lab and Isaac Sim and breaks backwards compatibility, but will come with many great fixes and improvements.

Full Changelog: https://github.com/isaac-sim/IsaacLab/compare/v1.3.0...v1.4.0

✨ **New Features**
Adds Factory contact-rich manipulation tasks to IsaacLab by @noseworm in https://github.com/isaac-sim/IsaacLab/pull/1520
Adds a Franka stacking ManagerBasedRLEnv by @peterd-NV in https://github.com/isaac-sim/IsaacLab/pull/1494
Adds recorder manager in manager-based environments by @nvcyc in https://github.com/isaac-sim/IsaacLab/pull/1336
Adds Ray Workflow: Multiple Run Support, Distributed Hyperparameter Tuning, and Consistent Setup Across Local/Cloud by @glvov-bdai in https://github.com/isaac-sim/IsaacLab/pull/1301
Adds OperationSpaceController to docs and tests and implement corresponding action/action_cfg classes by @ozhanozen in https://github.com/isaac-sim/IsaacLab/pull/913
Adds null-space control option within OperationSpaceController by @ozhanozen in https://github.com/isaac-sim/IsaacLab/pull/1557
Adds observation term history support to Observation Manager by @jtigue-bdai in https://github.com/isaac-sim/IsaacLab/pull/1439
Adds live plots to managers by @pascal-roth in https://github.com/isaac-sim/IsaacLab/pull/893

🔧 **Improvements**
Adds documentation and example scripts for sensors by @mpgussert in https://github.com/isaac-sim/IsaacLab/pull/1443
Removes duplicated TerminationsCfg code in G1 and H1 RoughEnvCfg by @fan-ziqi in https://github.com/isaac-sim/IsaacLab/pull/1484
Adds option to change the clipping behavior for all Cameras and unifies the default by @pascal-roth in https://github.com/isaac-sim/IsaacLab/pull/891
Adds check that no articulation root API is applied on rigid bodies by @lgulich in https://github.com/isaac-sim/IsaacLab/pull/1358
Adds RayCaster rough terrain base height to reward by @Andy-xiong6 in https://github.com/isaac-sim/IsaacLab/pull/1525
Adds position threshold check for state transitions by @DorsaRoh in https://github.com/isaac-sim/IsaacLab/pull/1544
Adds clip range for JointAction by @fan-ziqi in https://github.com/isaac-sim/IsaacLab/pull/1476

🐛 **Bug Fixes**
Fixes noise_model initialized in direct_marl_env by @NoneJou072 in https://github.com/isaac-sim/IsaacLab/pull/1480
Fixes entry_point and kwargs in isaaclab_tasks README by @fan-ziqi in https://github.com/isaac-sim/IsaacLab/pull/1485
Fixes syntax for checking if pre-commit is installed in isaaclab.sh by @louislelay in https://github.com/isaac-sim/IsaacLab/pull/1422
Corrects fisheye camera projection types in spawner configuration by @command-z-z in https://github.com/isaac-sim/IsaacLab/pull/1361
Fixes actuator velocity limits propagation down the articulation root_physx_view by @jtigue-bdai in https://github.com/isaac-sim/IsaacLab/pull/1509
Computes Jacobian in the root frame inside the DifferentialInverseKinematicsAction class by @zoctipus in https://github.com/isaac-sim/IsaacLab/pull/967
Adds transform for mesh_prim of ray caster sensor by @clearsky-mio in https://github.com/isaac-sim/IsaacLab/pull/1448
Fixes configclass dict conversion for torch tensors by @lgulich in https://github.com/isaac-sim/IsaacLab/pull/1530
Fixes error in apply_actions method in NonHolonomicAction action term. by @KyleM73 in https://github.com/isaac-sim/IsaacLab/pull/1513
Fixes outdated sensor data after reset by @kellyguo11 in https://github.com/isaac-sim/IsaacLab/pull/1276
Fixes order of logging metrics and sampling commands in command manager by @Mayankm96 in https://github.com/isaac-sim/IsaacLab/pull/1352

💔 **Breaking Changes**
Refactors pose and velocities to link frame and COM frame APIs by @jtigue-bdai in https://github.com/isaac-sim/IsaacLab/pull/966

🤗 **New Contributors**
@nvcyc made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1336
@peterd-NV made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1494
@NoneJou072 made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1480
@clearsky-mio made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1448
@Andy-xiong6 made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1525
@noseworm made their first contribution in https://github.com/isaac-sim/IsaacLab/pull/1520
